### ISSUE TITLE
Training Encoder X-ray Dataset

### ISSUE THREAD

**Issue by sriderya** [eladrich/pixel2style2pixel#142]
Hello,

First of all, thanks for your great work. I am trying to train the encoder on chest X-ray dataset. Although the results seems good, some details are missing, especially for medical sense. As can be seen from the example below, some important details such as cables are not recovered and this case is absolutely undesirable. By the way, results may seem pretty good for you but medical experts totally disagree :)

![0002_200000](https://user-images.githubusercontent.com/9121907/118790443-70199380-b89e-11eb-98e0-a5b35736ddb3.jpg)

The parameters are:

{
    "batch_size": 8,
    "board_interval": 50,
    "checkpoint_path": null,
    "d_reg_every": 16,
    "dataset_type": "xray_encode",
    "delta_norm": 2,
    "delta_norm_lambda": 0.0002,
    "encoder_type": "Encoder4Editing",
    "exp_dir": "/path/to/experiment/dir",
    "id_lambda": 0.5,
    "image_interval": 100,
    "keep_optimizer": false,
    "l2_lambda": 1.0,
    "learning_rate": 0.0001,
    "lpips_lambda": 0.8,
    "lpips_type": "alex",
    "max_steps": 200000,
    "optim_name": "ranger",
    "progressive_start": 20000,
    "progressive_step_every": 2000,
    "progressive_steps": [
        0,
        20000,
        22000,
        24000,
        26000,
        28000,
        30000,
        32000,
        34000,
        36000,
        38000,
        40000,
        42000,
        44000
    ],
    "r1": 10,
    "resume_training_from_ckpt": null,
    "save_interval": null,
    "save_training_data": false,
    "start_from_latent_avg": true,
    "stylegan_size": 256,
    "stylegan_weights": "/path/to/stylegan2.pt",
    "sub_exp_dir": null,
    "test_batch_size": 4,
    "test_workers": 4,
    "train_decoder": false,
    "update_param_list": null,
    "use_w_pool": true,
    "val_interval": 10000,
    "w_discriminator_lambda": 0.1,
    "w_discriminator_lr": 2e-05,
    "w_pool_size": 50,
    "workers": 8
}

In order to get better inversion for this kind of dataset, which parameters should I tune? How can I improve my results?

Thanks in advance

**Comment by yuval-alaluf** [comment]
Hi @sriderya 
It seems like you are training using e4e but posted on the pSp repo :) 
In any cases e4e is known to trade off reconstruction quality for editability, which as you mentioned is not desirable in your case. 
I recommend trying to train using pSp using the moco loss we incorporated a month ago (more details are provided in the Readme, but let me know if you have more questions).
If this still leads to undesirable results, I would recommend using my more recent ReStyle encoder which I showed to outperform both pSp and e4e by a good margin (code is also available and is based on pSp's codebase so running ReStyle should be straightforward)

**Comment by sriderya** [comment]
Thanks @yuval-alaluf for your answer. I am sorry for my terrible mistake, what was I thinking :)
I will try your suggestions immediately and will give feedback, thanks a lot.

**Comment by mmiaz** [comment]
@sriderya Hi, amazing work. May I know did you train your own styleGAN model as well?
